.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -1,19 +1,24 @@
-Add compact, short information about your PR for easier understanding:
+<!-- Add compact, short information about your PR for easier understanding, like the objective of the PR -->
 
-- Goal of the PR
-- How does the PR work?
-- Does it resolve any reported issue?
-- If not a bug fix, why is this PR needed? What usecases does it solve?
+### How does the PR work?
+<!-- Elaborate on the changes, if applicable, or delete this heading -->
 
-## To do
+### Use-cases
+<!-- What use cases does this PR serve? Delete this heading if the PR does not implement new features -->
 
-This PR is a Work in Progress / Ready for Review.
-<!-- ^ delete one -->
+****
 
-- [ ] List
```

Comment:
I'd still encourage the classic to-do list due to the good GH integration. So maybe leave one `- [ ] TODO` line there?

Reply:
Most open PRs remove the checklist altogether (I checked :P). But I see why you think it's a good idea to include it in the template. I'll mention it in a comment instead, given that it's mostly removed.